modules/lazyload/module.php

```diff
@@ -64,21 +65,46 @@ private function update_element_attributes( Element_Base $element ) {
 	}
 
 	private function append_lazyload_selector( $control, $value ) {
+		$is_dominant_color_enabled = $this->is_dominant_color_enabled();
+
 		if ( Utils::get_array_value_by_keys( $control, [ 'background_lazyload', 'active' ] ) ) {
 			foreach ( $control['selectors'] as $selector => $css_property ) {
 				if ( 0 === strpos( $css_property, 'background-image' ) ) {
 					if ( ! empty( $value['url'] ) ) {
 						$css_property  = str_replace( 'url("{{URL}}")', 'var(--e-bg-lazyload-loaded)', $css_property );
 						$control['selectors'][ $selector ] = $css_property . '--e-bg-lazyload: url("' . $value['url'] . '");';
-						$control = $this->apply_dominant_color_background( $control, $value, $selector );
+
+						if ( $is_dominant_color_enabled ) {
+							$control = $this->apply_dominant_color_background( $control, $value, $selector );
+						}
 					}
 				}
 			}
 		}
 		return $control;
 	}
 
+	private function is_dominant_color_enabled() {
```

Comment:
It feels like the code should not be inside the lazy load module, maybe it should not be in Elementor at all and we should provide a hook to the `PERFORMANCE_LAB` team, and they will add this code.

Reply:
PERFORMANCE_LAB does not know about our CSS or background images, we have our own post.css for each page or post. Our controls and CSS are not parsed by them.

We use an internal CSS mechanism to create the background image & all the page css. Previously, we had a problem with this plugin and our CSS, so I created a module for it.

https://github.com/elementor/elementor/blob/ef19421b7637f93d427e1691427e8e87304015a7/modules/performance-lab/module.php